lib/matplotlib/colorbar.py

```diff
@@ -670,35 +656,35 @@ def _do_extends(self, extendlen):
         if self._extend_lower():
             if not self.extendrect:
                 # triangle
-                xy = np.array([[0.5, 0], [1, elower], [0, elower]])
+                xy = np.array([[0.5, -elower], [1, 0], [0, 0]])
             else:
                 # rectangle
-                xy = np.array([[0, 0], [1., 0], [1, elower], [0, elower]])
+                xy = np.array([[0, -elower], [1., -elower], [1, 0], [0, 0]])
             if self.orientation == 'horizontal':
                 xy = xy[:, ::-1]
             # add the patch
             color = self.cmap(self.norm(self._values[0]))
             patch = mpatches.PathPatch(
                 mpath.Path(xy), facecolor=color, linewidth=0,
-                antialiased=False, transform=self.ax.outer_ax.transAxes,
-                hatch=hatches[0])
-            self.ax.outer_ax.add_patch(patch)
+                antialiased=False, transform=self.ax.transAxes,
+                hatch=hatches[0], clip_on=False)
+            self.ax.add_patch(patch)
```

Comment:
Because you're setting `clip_on=False` here, do you need to adjust the zorder of the patch at all so that it is behind the main axes?

Reply:
I think this is as it was before?

Reply:
This seems reasonable with the layer between the patch, the outline, and then the fill-ins for the extensions.

We do not keep a (named) reference to these patches, what do we do if we update the colormap?